packages/flutter_tts/tizen/src/text_to_speach.cc

```diff
@@ -0,0 +1,159 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "text_to_speach.h"
+
+#include "log.h"
+
+namespace {
+
+void OnStateChanged(tts_h tts, tts_state_e previous, tts_state_e current,
+                    void *user_data) {
+  TextToSpeach *self = static_cast<TextToSpeach *>(user_data);
+  self->OnStateChanged(previous, current);
+}
+
+void OnUtteranceCompleted(tts_h tts, int utt_id, void *user_data) {
+  TextToSpeach *self = static_cast<TextToSpeach *>(user_data);
+  self->OnUtteranceCompleted(utt_id);
+  // Explicitly call stop method to change the tts state to ready.
+  self->Stop();
+}
+
+void OnError(tts_h tts, int utt_id, tts_error_e reason, void *user_data) {
+  TextToSpeach *self = static_cast<TextToSpeach *>(user_data);
+  self->OnError(utt_id, reason);
+}
+
+}  // namespace
+
+void TextToSpeach::Init() {
+  int ret = tts_create(&tts_);
+  if (ret != TTS_ERROR_NONE) {
+    tts_ = nullptr;
+    LOG_ERROR("[TTS] tts_create failed: %s", get_error_message(ret));
+    // TODO : throw error
```

Comment:
Do you plan to finish this and TODOs in later commits?

Reply:
For now, I don't have any plan.actually, this is close to writing my thoughts I had during development.
If you wish, I'll keep the PR open and do some further work for this but If you don't, I'll do it later

Reply:
I'm okay with doing it later. Then could you add "TODO" in the last comment as well so that it's easy to search in the future?
Also could you leave some details in the comments so that others may pick up without too much effort? Did you mean sending error code to the dart side?
Does the "check return and throw error" mean: check return values of each set methods below and send error code to dart side on failure?

Reply:
Please, review this [commit](https://github.com/flutter-tizen/plugins/pull/157/commits/2a2ea6f39b7b4f52037db7c20a37c949b57c71b2).